### PR TITLE
fix: validate config after profile selection

### LINKED ISSUE #5777: init deploys with default profile if the selected profile is invalid
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
When creating a new project using `amplify init`, if the profile credentials defined in ~/.aws/credentials are invalid, the Amplify CLI does not catch this error and proceeds init with a (random?) profile

**Amplify CLI Version**
4.32.0

**To Reproduce**
1. Modify ~/.aws/credentials for the intended profile, modify key `aws_secret_access_key` to read `aws_secret-access_key`.
2. Validate aws cli is capable of handling the error
```
alex@DESKTOP-6H0TC0U:~/repos/it-inventory/inventory$ aws sts get-caller-identity --profile broken
Partial credentials found in shared-credentials-file, missing: aws_secret_access_key
```
3. Init a new amplify project
```
alex@DESKTOP-6H0TC0U:~/repos/it-inventory/inventory$ amplify init
Note: It is recommended to run this command from the root of your app directory
? Enter a name for the project inventory
? Enter a name for the environment dev
? Choose your default editor: Visual Studio Code
? Choose the type of app that you're building javascript
Please tell us about your project
? What javascript framework are you using vue
? Source Directory Path:  src
? Distribution Directory Path: dist
? Build Command:  npm run-script build
? Start Command: npm run-script serve
Using default provider  awscloudformation

For more information on AWS Profiles, see:
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html

? Do you want to use an AWS profile? Yes
? Please choose the profile you want to use broken
Adding backend environment dev to AWS Amplify Console app: d11yelff4xoqvu
⠇ Initializing project in the cloud...

CREATE_IN_PROGRESS UnauthRole                   AWS::IAM::Role             Wed Nov 04 2020 20:51:22 GMT-0600 (Central Standard Time) Resource creation Initiated
CREATE_IN_PROGRESS AuthRole                     AWS::IAM::Role             Wed Nov 04 2020 20:51:22 GMT-0600 (Central Standard Time) Resource creation Initiated
CREATE_IN_PROGRESS UnauthRole                   AWS::IAM::Role             Wed Nov 04 2020 20:51:21 GMT-0600 (Central Standard Time)
CREATE_IN_PROGRESS DeploymentBucket             AWS::S3::Bucket            Wed Nov 04 2020 20:51:21 GMT-0600 (Central Standard Time)
CREATE_IN_PROGRESS AuthRole                     AWS::IAM::Role             Wed Nov 04 2020 20:51:21 GMT-0600 (Central Standard Time)
CREATE_IN_PROGRESS amplify-inventory-dev-205111 AWS::CloudFormation::Stack Wed Nov 04 2020 20:51:17 GMT-0600 (Central Standard Time) User Initiated
⠋ Initializing project in the cloud...

CREATE_IN_PROGRESS DeploymentBucket AWS::S3::Bucket Wed Nov 04 2020 20:51:23 GMT-0600 (Central Standard Time) Resource creation Initiated
⠦ Initializing project in the cloud...

CREATE_COMPLETE UnauthRole AWS::IAM::Role Wed Nov 04 2020 20:51:38 GMT-0600 (Central Standard Time)
CREATE_COMPLETE AuthRole   AWS::IAM::Role Wed Nov 04 2020 20:51:38 GMT-0600 (Central Standard Time)
⠏ Initializing project in the cloud...

CREATE_COMPLETE amplify-inventory-dev-205111 AWS::CloudFormation::Stack Wed Nov 04 2020 20:51:46 GMT-0600 (Central Standard Time)
CREATE_COMPLETE DeploymentBucket             AWS::S3::Bucket            Wed Nov 04 2020 20:51:44 GMT-0600 (Central Standard Time)
✔ Successfully created initial AWS cloud resources for deployments.
✔ Initialized provider successfully.
Initialized your environment successfully.

Your project has been successfully initialized and connected to the cloud!
```

After some digging, I determined by observing the ARNs listed in `amplify/cli.json` that the cloudformation template was deployed using a profile I did not select.

**Expected behavior**
Cloudformation template should deploy using indicated profile

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 in WSLv2
 - Node Version. 14.4.0

### PR DESCRIPTION
**Pull request by nikhname**
fix #5777

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by yuth — CHANGES_REQUESTED** [review]
This handles profile validation only in case of init. If the prole is missing then `amplify push` will end up using the default profile. Please update the code to throw errors when the named profile is missing.

File: packages/amplify-provider-awscloudformation/src/system-config-manager.js [hunk 1]
```diff
@@ -286,7 +286,22 @@ function getProfileCredentials(profileName) {
...
+  profileCredentials = normalizeKeys(profileCredentials);
+  validateCredentials(profileCredentials, profileName);
+  return profileCredentials;
+}
+
+function validateCredentials(credentials, profileName) {
+  if (credentials && credentials.accessKeyId === undefined) {
+    const err = new Error(`Profile configuration is invalid for: ${profileName}, missing aws_access_key_id`);
+    logger('validateCredentials', [profileName])(err);
+    throw err;
```
Comment (attilah) [inline packages/amplify-provider-awscloudformation/src/system-config-manager.js:298]: Add `err.stack = undefined;` so we'll not print stacktrace and that's a better DX for customers.

**Review by attilah — CHANGES_REQUESTED** [review]
Update based on comments pleased.

**Comment by nikhname** [comment]
> This handles profile validation only in case of init. If the prole is missing then `amplify push` will end up using the default profile. Please update the code to throw errors when the named profile is missing.

Is this true? It looks like if the profile is missing altogether we error out at this [line](https://github.com/aws-amplify/amplify-cli/blob/7d7507e593b22921b71382f8d0558fd0b8593300/packages/amplify-provider-awscloudformation/src/system-config-manager.js#L90) in both init and push.

**Comment by yuth** [comment]
> > This handles profile validation only in case of init. If the prole is missing then `amplify push` will end up using the default profile. Please update the code to throw errors when the named profile is missing.
> 
> Is this true? It looks like if the profile is missing altogether we error out at this [line](https://github.com/aws-amplify/amplify-cli/blob/7d7507e593b22921b71382f8d0558fd0b8593300/packages/amplify-provider-awscloudformation/src/system-config-manager.js#L90) in both init and push.

I still can reproduce this. In am already provisioned project go and change the profile name to 'some-random-which-doesnt-exists` in `amplify/.config/local-aws-info.json` and do an push and it still goes through.

**Comment by jhockett** [comment]
Closing in favor of https://github.com/aws-amplify/amplify-cli/pull/7017

**Pull request closed without merging**